target/ppc/translate/vsx-impl.c.inc

```diff
@@ -2140,48 +2140,46 @@ static bool do_lstxv(DisasContext *ctx, arg_D *a, bool store, bool paired)
     ea = tcg_temp_new();
 
     mop = DEF_MEMOP(MO_Q);
-    le = ctx->le_mode;
-
-    do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si + ((paired && le)*16)), ea);
 
     gen_set_access_type(ctx, ACCESS_INT);
 
     xt = tcg_temp_new_i64();
 
-    /*
-     * If big-endian vsr64_offset(n, !le) returns vsx register first 64bits,
-     * otherwise returns last 64bits.
-     * If big-endian vsr64_offset(n, le) returns vsx register last 64bits,
-     * otherwise returns first 64bits.
-     */
+    if (ctx->le_mode) {
+        do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si + (paired ? 24 : 8)), ea);
+        offset = -8;
+    } else {
+        do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si), ea);
+        offset = 8;
+    }
 
     if(store) {
-        tcg_gen_ld_i64(xt, cpu_env, vsr64_offset(a->rt, !le));
+        get_cpu_vsrh(xt, a->rt);
         tcg_gen_qemu_st_i64(xt, ea, ctx->mem_idx, mop);
-        tcg_gen_addi_tl(ea, ea, 8);
-        tcg_gen_ld_i64(xt, cpu_env, vsr64_offset(a->rt, le));
+        tcg_gen_addi_tl(ea, ea, offset);
```

Comment:
You need to replace any ``tcg_gen_addi_tl(ea, ea, offset);`` with ``gen_addr_add(ctx, ea, ea, offset);``, which does the NARROW_MODE handling after each ``ea`` change.

Reply:
Changed to gen_addr_add as suggested